src/lib.rs

```diff
@@ -61,38 +61,34 @@
     html_logo_url = "https://raw.githubusercontent.com/smol-rs/smol/master/assets/images/logo_fullsize_transparent.png"
 )]
 
-use std::convert::TryFrom;
 use std::future::Future;
-use std::io::{self, IoSlice, IoSliceMut, Read, Write};
-use std::net::{SocketAddr, TcpListener, TcpStream, UdpSocket};
 use std::pin::Pin;
-use std::sync::Arc;
-use std::task::{Context, Poll, Waker};
-use std::time::{Duration, Instant};
+use std::task::{Context, Poll};
+use std::time::Duration;
 
-#[cfg(unix)]
-use std::{
-    os::unix::io::{AsFd, AsRawFd, BorrowedFd, OwnedFd, RawFd},
-    os::unix::net::{SocketAddr as UnixSocketAddr, UnixDatagram, UnixListener, UnixStream},
-    path::Path,
-};
+#[cfg(not(target_family = "wasm"))]
+use std::time::Instant;
 
-#[cfg(windows)]
-use std::os::windows::io::{AsRawSocket, AsSocket, BorrowedSocket, OwnedSocket, RawSocket};
-
-use futures_io::{AsyncRead, AsyncWrite};
-use futures_lite::stream::{self, Stream};
-use futures_lite::{future, pin, ready};
-use socket2::{Domain, Protocol, SockAddr, Socket, Type};
-
-use crate::reactor::{Reactor, Registration, Source};
+use futures_lite::stream::Stream;
 
+#[cfg(not(target_family = "wasm"))]
 mod driver;
+#[cfg(not(target_family = "wasm"))]
+mod io;
+#[cfg(not(target_family = "wasm"))]
 mod reactor;
 
+#[cfg_attr(not(target_family = "wasm"), path = "timer/native.rs")]
+#[cfg_attr(target_family = "wasm", path = "timer/web.rs")]
```

Comment:
This is actually only necessary in `target_os = "unknown"` cases, right?

Reply:
WASI doesn't have support in `polling` yet... although WASI also doesn't really have support for web primitives either, so compiling to it should really be a compile error.

As for `emscripten`, I don't know whether it supports `poll` or `epoll` or anything, so until we figure that out we should be using web timers.

So you're right, this should be `any(target_os = "unknown", target_os = "emscripten")`.

Reply:
....actually I'm pretty sure that `not(target_os = "wasi")` covers our bases better.